swiftnav/src/time/gnss.rs

```diff
@@ -0,0 +1,816 @@
+    /// preferable to use [`GpsTime::to_utc()`] with the newest set of UTC parameters
+    ///
+    /// # Panics
+    ///
+    /// This function will panic if the GPS time is not valid
+    pub fn to_utc_hardcoded(self) -> UtcTime {
+        self.internal_to_utc(None)
+    }
+
+    /// Gets the number of seconds difference between GPS and UTC times
+    pub(crate) fn gps_utc_offset(&self, utc_params: &UtcParams) -> f64 {
+        let dt = self.diff(&utc_params.tot());
+
+        /* The polynomial UTC to GPS correction */
+        let mut dt_utc: f64 =
+            utc_params.a0() + (utc_params.a1() * dt) + (utc_params.a2() * dt * dt);
+
+        /* the new UTC offset takes effect after the leap second event */
+        if self.diff(&utc_params.t_lse()) >= 1.0 {
+            dt_utc += utc_params.dt_lsf() as f64;
+        } else {
+            dt_utc += utc_params.dt_ls() as f64;
+        }
+
+        dt_utc
+    }
+
+    /// Gets the number of seconds difference between GPS and UTC using the hardcoded
+    /// list of leap seconds
+    ///
+    /// # ⚠️  🦘  ⏱  ⚠️  - Leap Seconds
+    ///
+    /// The hard coded list of leap seconds will get out of date, it is
+    /// preferable to use [`GpsTime::gps_utc_offset()`] with the newest set
+    /// of UTC parameters
+    pub(crate) fn gps_utc_offset_hardcoded(&self) -> f64 {
+        for (t_leap, offset) in UTC_LEAPS.iter().rev() {
+            if self.diff(t_leap) >= 1.0 {
+                return *offset;
+            }
+        }
+
+        /* time is before the first known leap second event */
+        0.0
+    }
+
+    /// Gets the number of seconds difference between UTC and GPS using the hardcoded
+    /// list of leap seconds
+    pub(crate) fn utc_gps_offset(&self, utc_params: &UtcParams) -> f64 {
+        let dt = self.diff(&utc_params.tot()) + utc_params.dt_ls() as f64;
+
+        /* The polynomial UTC to GPS correction */
+        let mut dt_utc = utc_params.a0() + utc_params.a1() * dt + utc_params.a2() * dt * dt;
+
+        /* the new UTC offset takes effect after the leap second event */
+        if self.diff(&utc_params.t_lse()) >= ((-utc_params.dt_ls() as f64) - dt_utc) {
+            dt_utc += utc_params.dt_lsf() as f64;
+        } else {
+            dt_utc += utc_params.dt_ls() as f64;
+        }
+
+        -dt_utc
+    }
+
+    /// Gets the number of seconds difference between UTC and GPS using the hardcoded
+    /// list of leap seconds
+    ///
+    /// # ⚠️  🦘  ⏱  ⚠️  - Leap Seconds
+    /// The hard coded list of leap seconds will get out of date, it is
+    /// preferable to use [`GpsTime::utc_gps_offset()`] with the newest set
+    /// of UTC parameters
+    pub(crate) fn utc_gps_offset_hardcoded(&self) -> f64 {
+        for (t_leap, offset) in UTC_LEAPS.iter().rev() {
+            if self.diff(t_leap) >= (-offset + 1.0) {
+                return -offset;
+            }
+        }
+
+        /* time is before the first known leap second event */
+        0.0
```

Comment:
I'll stop suggesting this, but, if you want, you don't have to make all of these like `0.` or `0.0`, you can just do `0`